### PR TITLE
Script for uploading to artifactory

### PR DESCRIPTION
**Pull request by AAndrisa**
This script is used for a release job which takes the artifacts from azure build and then upload the files to artifactory sever.

## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: tools/scripts/upload_to_artifactory.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# SPDX-License-Identifier: (GPL-1.0-only OR BSD-2-Clause)
```
Comment (amiclaus) [inline tools/scripts/upload_to_artifactory.py:1]: was this script copied from the linux repo? we don't have such Licenses on no-os repository, as far as I know.

File: tools/scripts/upload_to_artifactory.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+########################################################################
+#
+# File name: upload_to_artifactory.py
+# Author: Raus Stefan
```
Comment (amiclaus) [inline tools/scripts/upload_to_artifactory.py:6]: is the author still valid?

File: tools/scripts/upload_to_artifactory.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+#
+# This script is used to upload files to ADI internal artifactory server
+#
+# Copyright (C) 2019-2022 Analog Devices Inc.
```
Comment (amiclaus) [inline tools/scripts/upload_to_artifactory.py:10]: 2023?

File: tools/scripts/upload_to_artifactory.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+#
+#######################################################################
+
+#!/usr/bin/python3.6
```
Comment (amiclaus) [inline tools/scripts/upload_to_artifactory.py:14]: is the script constrained to python3.6?

File: tools/scripts/upload_to_artifactory.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+if args.base_path:
+   UPLOAD_BASE_PATH = args.base_path
+else:
+   if "UPLOAD_BASE_PATH" in os.environ:
+      UPLOAD_BASE_PATH = os.environ['UPLOAD_BASE_PATH']
+   else:
+      print('\nError:Parameter "--base_path" is not set. This is ADI Internal Artifactory Server plus first level of folders. It can be set even using parameter "--base_path" on upload command, even by exporting UPLOAD_BASE_PATH variable in terminal, before calling upload script.')
+      exit(1)
+
```
Comment (amiclaus) [inline tools/scripts/upload_to_artifactory.py:67]: extra line

**Review by amiclaus — COMMENTED** [review]
some comments on my side. Also I checked the usage in the Azure Pipeline Releases and I have some queries:
1. Why creating separated Release pipeline for uploading to artifactory and not creating a extra task in the `Noos_latest_projects` pipeline? (or even include it in the existing Bash script - `release_projects.sh`)
image
2. I checked the new Pipeline and it seems that we use inline Bash Scripting. Why not adding the script here in the no-OS repo? (similar to https://github.com/analogdevicesinc/no-OS/blob/master/tools/scripts/release_projects.sh) and add comments in the script on how to use it in Azure Pipelines (i.e. https://github.com/analogdevicesinc/no-OS/blob/e12dd1e7a6cec1862133a6d8459820f40bb4d5f9/tools/scripts/release_projects.sh#L36-L42).
The inline Bash Script used directly in the Pipelines can be easily lost/deleted. We need to keep things in one place (and documented) as much as possible.

**Comment by AAndrisa** [comment]
V2: resolved the comments from PR, removing python version and licenses, also modify author and year from header.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**